packages/toolpad-app/src/server/localMode.ts

```diff
@@ -964,6 +966,22 @@ export function getRequiredEnvVars(dom: appDom.AppDom): Set<string> {
   return new Set(allVars);
 }
 
+export interface PaidFeaturesConfig {
+  roles?: boolean;
+  'role permissions'?: boolean;
+}
+
+export function detectPaidFeatures(application: Application): PaidFeaturesConfig | null {
+  if (!application.spec || !application.spec.authorization) {
+    return null;
+  }
+
+  const { authorization } = application.spec;
+  const hasRoles = authorization.roles && authorization.roles.length > 0;
```

Comment:
Shouldn't have the roles property at all IMO. 

```suggestion
  const hasRoles = !!application?.spec?.authorization?.roles
```